src/coreclr/jit/emitxarch.cpp

```diff
@@ -478,70 +478,162 @@ bool emitter::IsFlagsAlwaysModified(instrDesc* id)
 //    true if previous instruction zeroed reg's upper 32 bits.
 //    false if it did not, or if we can't safely determine.
 //
-// Notes:
-//    Currently only looks back one instruction.
-//
-//    movsx eax, ... might seem viable but we always encode this
-//    instruction with a 64 bit destination. See TakesRexWPrefix.
-
 bool emitter::AreUpper32BitsZero(regNumber reg)
 {
+#ifdef TARGET_AMD64
+    if (!((reg >= REG_RAX) && (reg <= REG_XMM15)))
+        return false;
+#else
+    if (!((reg >= REG_EAX) && (reg <= REG_XMM7)))
+        return false;
+#endif // !TARGET_AMD64
```

Comment:
minor - I think that you could remove this `ifdef` by using `REG_FP_LAST`

Reply:
Yea, looks like I can use this, then I can use `REG_EAX` as it is just an alias for `REG_RAX`